File: library/src/main/java/org/xmtp/android/library/libxmtp/UnpublishedMessage.kt
package uniffi.xmtpv3.org.xmtp.android.library.libxmtp

import org.xmtp.android.library.toHex
import uniffi.xmtpv3.FfiUnpublishedMessage

class UnpublishedMessage(private val libXMTPUnpublishedMessage: FfiUnpublishedMessage) {
    val messageId: String
        get() = libXMTPUnpublishedMessage.id().toHex()

    suspend fun publish() {
        libXMTPUnpublishedMessage.publish()
    }
}

File: Sources/XMTPiOS/Mls/UnpublishedMessage.swift
//
//  UnpublishedMessage.swift
//  
//
//  Created by Naomi Plasterer on 7/8/24.
//

import Foundation
import LibXMTP

public struct UnpublishedMessage: Identifiable {
	let ffiUnpublishedMessage: FfiUnpublishedMessage
	
	init(ffiUnpublishedMessage: FfiUnpublishedMessage) {
		self.ffiUnpublishedMessage = ffiUnpublishedMessage
	}
	
	public var id: String {
		return ffiUnpublishedMessage.id().toHex
	}
	
	public func publish() async throws -> String {
		try await ffiUnpublishedMessage.publish()
		return id
	}
}
